### PR TITLE
Add source address to UAttributes

### PR DESCRIPTION
**Pull request by stevenhartley**
#101

### CODE REVIEW

**Review by MelamudMichael — APPROVED** [review]

File: uprotocol/uri.proto [hunk 1]
```diff
@@ -58,9 +58,11 @@ message UUri {
 // is local.
 message UAuthority {
   optional string name = 1; // Domain & device name as a string
-  optional bytes ip = 2; 		// IPv4 or IPv6 Address in byte format
-  optional bytes id = 3;		// Unique ID for the device, could be a VIN, SHA 128, or any other identifier
-                            // *NOTE:* MAX length is 255 bytes
+  oneof number {
```
Comment (PLeVasseur) [inline uprotocol/uri.proto:61]: Nice :hugs:

**Review by PLeVasseur — APPROVED** [review]
LGTM :+1:

File: uprotocol/uattributes.proto [hunk 2]
```diff
@@ -46,38 +46,43 @@ message UAttributes {
...
 
+    // The source (address) of the message. For published events this is the producers publish topic
+    // for requests it is the calling uE respose topic (who sent the request). For responses this is the
+    // method URI from the uService. 
+    UUri source = 3;
+
     // Destination UUri for a message used for unicast message types 
     // (notification, request, response)
-    optional UUri sink = 3;
+    optional UUri sink = 4;
```
Comment (sophokles73) [inline uprotocol/uattributes.proto:56]: My understanding is that in proto3 message typed properties are always optional.

Is there a particular reason why some of them are explicitly marked `optional` while others are not?

File: uprotocol/uattributes.proto [hunk 3]
```diff
@@ -46,38 +46,43 @@ message UAttributes {
...
     // has occurred in the delivery of RPC request or response events. 
     // The contents of this attribute, if present, is the unsigned integer representation of
     // UCode
-    optional int32 commstatus = 7;
+    optional int32 commstatus = 8;
 
     // The correlation ID (UUDI) passed for response type messages to corelate the 
     // response to a request.
-    optional UUID reqid = 8;
+    optional UUID reqid = 9;
```
Comment (sophokles73) [inline uprotocol/uattributes.proto:79]: `optional` keyword can be removed FMPOV

File: uprotocol/uattributes.proto [hunk 4]
```diff
@@ -46,38 +46,43 @@ message UAttributes {
     // Message type (publish, request, response)
     UMessageType type = 2;
 
+    // The source (address) of the message. For published events this is the producers publish topic
+    // for requests it is the calling uE respose topic (who sent the request). For responses this is the
+    // method URI from the uService. 
+    UUri source = 3;
```
Comment (sophokles73) [inline uprotocol/uattributes.proto:52]: I hope you [know what you are doing](https://protobuf.dev/programming-guides/dos-donts/#reuse-number), shifting the identifiers ;-)

File: uprotocol/uattributes.proto (hunk 2 above)
Comment (stevenhartley) [reply uprotocol/uattributes.proto]: Nope, removed you are correct.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Review by tamarafischer — APPROVED** [review]
Personally I don't think these changes are really needed and are driving the people actually using and writing code to the protocol to drink!
That said, what is another change in the everchanging land of software development.
<End of developer drama> 

That said, if everyone really feels this is needed you have my plus 1.

**Pull request merged**